app/controllers/tasks_controller.rb

```diff
@@ -4,42 +4,72 @@ def index
     render :index
   end
 
+  # Display a specific task
   def show
-    @task = Task.find(params[:id])
+    @task   = Task.find(params[:id])
     render :tasks
   end
 
   def new
     @task = Task.new
-
     render :new
   end
 
-  # Add and save a new task to db
+  # Add and save a new task to list
   def create
-    @task = Task.new(task_params[:task])
-    @task.save
-
+    task = Task.new(task_params[:task])
+    task.save
     index
   end
 
-  # Delete a task from db
+  # Delete a task from list
   def destroy
-    @task = Task.find(params[:id])
-    @task.destroy.save
-
+    task = Task.find(params[:id])
+    task.destroy.save
     redirect_to '/'
   end
 
+  # Update an existing task
+  def edit
+    @task = Task.find(params[:id])
+    render :update
+  end
+
   def update
     @task = Task.find(params[:id])
-    @task.update.save
+    @name = task_params[:task][:name]
+    @desc = task_params[:task][:desc]
+    @comp_date = task_params[:task][:comp_date]
+    @person_id = task_params[:task][:person_id]
+
+    @task.update(name: @name)
+    @task.update(desc: @desc)
+    @task.update(comp_date: @comp_date)
+    @task.update(person_id: @person_id)
```

Comment:
[`#update`](http://apidock.com/rails/ActiveRecord/Persistence/update) takes a hash so you could do all of these in a single call. 

Additionally, the keys to `#update` are the same as the keys in your parameters.  So you could simplify lines 40-48 as follows:

```ruby
@task.update(task_params[:task])
```

As written this will write to the database four times, which slows things down (and increases the likelihood that the data may change from another user between updates). Also, if one of these failed (because of validation or database connection error) the previous records would have completed and generally you want all of these in a single transaction.